flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/ReconciliationUtils.java

```diff
@@ -126,6 +126,9 @@ private static <SPEC extends AbstractFlinkSpec> void updateStatusForSpecReconcil
         status.setError(null);
         reconciliationStatus.setReconciliationTimestamp(clock.instant().toEpochMilli());
 
+        // Set observedGeneration
+        status.setObservedGeneration(target.getMetadata().getGeneration());
```

Comment:
I still don't think that this is consistent with the current content of the lastReconciledSpec metadata. It's not about setting it to the same value (source of truth) it's also about setting it at the exact same time.

Also we should remove internal useges of the lastReconciledSpec.meta.generation and replace it with this one. We have tests for the behaviour so that would validate it.

Reply:
@gyfora I see, thanks for the review. I moved the setting of `observedGeneration` to be at the same time/place as `lastReconciledSpec.meta.generation`, and replaced the internal usage of the latter in two locations. Tests pass, however please advise if there is somewhere I may have overlooked.